examples/ivr_outbound_dialer.py

```diff
@@ -0,0 +1,359 @@
+
+        self.stereo = False
+
+        # mod_sndfile module is a must in order to play prompts
+        # Example of how to execute FreeSWITCH/NSG commands as from the CLI
+        try:
+            client.cmd('load mod_sndfile')
+        except switchy.utils.CommandError:
+            pass
+
+        # Example of how to use a Timer to measure elapsed time
+        # This example does not use it, but its use is commented out and
+        # left for demonstration purposes
+        self.dtmf_timeout = switchy.utils.Timer()
+        self.dtmf_timeout_period = 3.0
+
+    @event_callback('CHANNEL_PARK')
+    def on_park(self, sess):
+        # Example of how to answer an inbound call
+        # This application demonstrates outbound dialing
+        if sess.is_inbound():
+            sess.answer()
+
+    @event_callback("CHANNEL_ANSWER")
+    def on_answer(self, sess):
+        call = sess.call
+
+        # This application does not deal with inbound calls.
+        if sess.is_inbound():
+            pass
+
+        # Outbound call hsa just been answered
+        # Developer would start the introductory IVR message
+        if sess.is_outbound():
+
+            # Start recording a call
+            call.vars['record'] = True
+            sess.start_record('{}/callee_{}.wav'.format(self.recdir, sess.uuid), stereo=self.stereo)
+
+            # Play IVR initial greeting
+            call.vars['play_welcome'] = True
+            play_filename = '{}/en/us/callie/ivr/8000/ivr-welcome.wav'.format(self.sound_dir)
+            sess.playback(play_filename)
+
+            # At this point we wait for PLAYBACK_STOP event to finish
+            # We could trigger a timeout as in DTMF section
+
+    # Timer handler function that implements DTMF timeout
+    def dtmf_timeout_action(self, sess):
+        call = sess.call
+
+        log.info("'{}': DTMF timeout".  format(sess.uuid))
+
+        if call.vars.get('playing') is True:
+            call.vars['playing'] = False
+            sess.breakmedia()
+
+        # Reset incoming dtmf queue
+        call.vars['incoming_dtmf'] = None
+
+        # Example of playing a prompt urging the user to try again
+        play_filename = '{}/en/us/callie/ivr/8000/ivr-hello.wav'.format(self.sound_dir)
+        call.vars['playing'] = True
+        sess.playback(play_filename)
+
+        # Trigger dtmf timeout again
+        call.vars['dtmf_timeout_job'] = threading.Timer(3, self.dtmf_timeout_action, [self, sess])
+
+    @event_callback('DTMF')
+    def on_digit(self, sess):
+        call = sess.call
+        digit = int(sess['DTMF-Digit'])
+
+        # DTMF has just been detected, stop playing any files to the user
+        if call.vars.get('playing') is True:
+            sess.breakmedia()
+
+        # If DTMF timeout is pending, reset the DTMF timeout
+        if call.vars.get('dtmf_timeout_job'):
+            log.debug("'{}': Cancel dtmf timeout job" . format(sess.uuid))
```

Comment:
I like the space between the dot.  It's more readable.
flake8 does not complain so it's up to the author :)